waku/nwaku_test.go

```diff
@@ -0,0 +1,396 @@
+package waku
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"testing"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/cenkalti/backoff/v3"
+	"github.com/libp2p/go-libp2p/core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
+	"github.com/waku-org/go-waku/waku/v2/protocol/store"
+)
+
+// In order to run these tests, you must run an nwaku node
```

Comment:
I updated the comment to be more accurate, because that separate waku node is only needed in for `TestBasicWaku`

In that test, the availability check we get from the call to `GetNwakuInfo`, which basically makes a REST API call to the node.

In the other tests (which only use the bindings) the current check is seeing if it runs `start()` without errors and then the following checks in each test make sure that everything runs as it should